src/objects.c

```diff
@@ -3360,7 +3359,68 @@ void unset_temp_model_transforms(Object *obj) {
     }
 }
 
-#pragma GLOBAL_ASM("asm/nonmatchings/objects/func_800135B8.s")
+// Renders the boost graphics.
+void func_800135B8(Object *boostObj) {
+    Vertex *vtx;
+    Triangle *tri;
+    ObjectTransform_800135B8 objTransform;
+    Object_Boost_Inner *boostData;
+    Object_Boost *boost;
+    UnkAsset_800135B8 *asset;
+    s32 hasTexture;
+    s32 idx;
+
+    idx = (boostObj->properties.common.unk4 >> 0x1C) & 0xF;
+    boost = &boostObj->unk64->boost;
+    switch (D_8011B048[idx]) {
+        case 0:
+            boostData = &boost->unk0;
+            break;
+        case 1:
+            boostData = &boost->unk24;
+            break;
+        default:
+            boostData = &boost->unk48;
+            break;
+    }
+    asset = (UnkAsset_800135B8 *) (get_misc_asset(20) + ((D_8011B058[idx] & 0xFF & 0xFF & 0xFF & 0xFF & 0xFF) << 5));
```

Comment:
I would add a comment here for the fake `& 0xFF & 0xFF & 0xFF & 0xFF & 0xFF`. Just for the random person who happens to look at a function here and go "wtf?"

Reply:
Turns out there was an even  better version without the fake `& 0xFF` by "properly" indexing asset by just using `D_8011B058[idx]` directly so I changed that